core/src/main/java/tc/oc/pgm/cycle/CycleMatchModule.java

```diff
@@ -71,4 +73,23 @@ public void onMatchEnd(MatchFinishEvent event) {
   public void onLeave(PlayerLeaveMatchEvent event) {
     if (bossbar != null) event.getPlayer().hideBossBar(bossbar);
   }
+
+  private Duration getCycleTime(MapOrder order) {
+    Duration cycleTime = null;
+
+    // Check if pool has custom cycle time
+    if (order instanceof MapPoolManager) {
+      MapPoolManager manager = (MapPoolManager) order;
+      if (manager != null && manager.getActiveMapPool() != null) {
+        cycleTime = manager.getActiveMapPool().getCycleTime();
+      }
+    }
+
+    // Default to the main cycle time when not found
+    if (cycleTime == null || cycleTime.isNegative()) {
+      cycleTime = PGM.get().getConfiguration().getCycleTime();
+    }
+
+    return cycleTime;
+  }
```

Comment:
Just move this method over to the MapOrder interface, have it default to `PGM.get().getConfiguration().getCycleTime();` and let implementations override the method.

Doing the if-hell of checking for specifically one implementation is a very bad practice since it does not allow for any other 3rd party implementation except for MapPoolManager to ever implement this feature

Reply:
Thanks for the feedback @Pablete1234 ! I've rewritten this to be more flexible for 3rd party implementations 👍 Let me know if you have any other suggestions.